gravitee-exchange-controller/gravitee-exchange-controller-websocket/src/main/java/io/gravitee/exchange/controller/websocket/WebSocketRequestHandler.java

```diff
@@ -56,49 +56,56 @@ public class WebSocketRequestHandler implements io.vertx.core.Handler<io.vertx.r
     public void handle(final RoutingContext routingContext) {
         log.debug("Incoming connection on Websocket Controller");
         HttpServerRequest request = routingContext.request();
-        ControllerCommandContext controllerContext = controllerAuthentication.authenticate(request);
-        if (controllerContext.isValid()) {
-            // Resolve protocol version from header
-            String headerValue = request.getHeader(EXCHANGE_PROTOCOL_HEADER);
-            ProtocolVersion protocolVersion = ProtocolVersion.parse(headerValue);
 
-            request
-                .toWebSocket()
-                .flatMapCompletable(webSocket -> {
-                    List<CommandHandler<? extends Command<?>, ? extends Reply<?>>> commandHandlers =
-                        controllerCommandHandlersFactory.buildCommandHandlers(controllerContext);
-                    List<CommandAdapter<? extends Command<?>, ? extends Command<?>, ? extends Reply<?>>> commandAdapters =
-                        controllerCommandHandlersFactory.buildCommandAdapters(controllerContext, protocolVersion);
-                    List<ReplyAdapter<? extends Reply<?>, ? extends Reply<?>>> replyAdapters =
-                        controllerCommandHandlersFactory.buildReplyAdapters(controllerContext, protocolVersion);
+        request.pause();
```

Comment:
I need this to avoid `Request has already been read` error. I've resumed the thing at [L108](https://github.com/gravitee-io/gravitee-exchange/pull/60/files#diff-0f68fa4338ac6e4536dbd3fd8d51de197eb33adaab7280fed46d74782a3eaea5R108) but maybe there is a cleaner way 🤔

Reply:
From what I understand https://github.com/vert-x3/vertx-web/issues/1908#issuecomment-1079849714 it is the expected behavior 🤷